### PR TITLE
Mdas 516 sst management c optimierung frontend

### PR DESCRIPTION
**Pull request by sfi2022**
## Pull Request
SST-Management Optimierung (Frontend)

### Description
Implements some refactoring issues described in MDAS-516
To view published to [dev](https://github.com/it-at-m/mobidam-sst-management/actions/runs/11048930460)

### Reference
MDAS-516

Issues:
See acceptance criteria MDAS-516.

### Definition of Done
*Advice: Unchecked checkboxes are not accepted by the build pipeline. Cross out any non-relevant item by using '~~'.*

- [x] Acceptance criteria are met
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification)
- [x] Frontend is locally smoke-tested
- [x] Board is updated
- [x] Infrastructure is adjusted

Please delete all side branches after merging.

### CODE REVIEW

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 1]
```diff
@@ -61,12 +61,13 @@
                     <v-text-field
                         ref="address"
                         v-model="zuordnung.funktionsadresse"
-                        label="Funktionsadresse"
+                        label="EMail"
                         hint="Welchem Gruppenpostfach gehört diese Person an?"
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:65]: Funktionspostfach

File: mobidam-sst-management-frontend/frontend/src/views/SchnittstelleDetail.vue [hunk 2]
```diff
@@ -106,26 +103,77 @@
...
+                                cols="3"
+                                md="3"
+                                sm="3"
+                                v-bind="props"
+                            >
+                                <v-icon>mdi-email</v-icon>
+                                {{ zuordnung.funktionsadresse }}</v-col
+                            >
+                        </template>
+                        Gruppenpostfach
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/views/SchnittstelleDetail.vue:144]: Funktionspostfach

File: mobidam-sst-management-frontend/frontend/src/views/MainView.vue [hunk 3]
```diff
@@ -61,7 +61,8 @@
                     <v-tooltip location="top">
                         <template #activator="{ props }">
                             <v-col
-                                cols="2"
+                                cols="3"
+                                md="3"
                                 v-bind="props"
                             >
                                 {{ schnittstelle.name }}
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/views/MainView.vue:68]: grafik
Theres plenty of room at the end of the table which could be better used for the sst-name

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 4]
```diff
@@ -61,12 +61,13 @@
                     <v-text-field
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:61]: "Benutzerkennung" instead of "Person angeben"

Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:61]: ✅

File: mobidam-sst-management-frontend/frontend/src/views/SchnittstelleDetail.vue (hunk 2 above)
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/views/SchnittstelleDetail.vue:144]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue (hunk 1 above)
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:65]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 5]
```diff
@@ -196,9 +208,36 @@ const form = ref<VForm>();
...
+                )
+            ) {
+                emit("zuordnung-saved", zuordnung.value);
+                closeDialog();
+            }
+        }
+    }
+}
+
+function isGueltigAbBeforeGueltigBis(
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:227]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 6]
```diff
@@ -112,8 +113,15 @@
...
                                         readonly
                                         variant="outlined"
-                                        :rules="textInputRules"
+                                        clearable
                                         v-bind="props"
+                                        :rules="[
+                                            isGueltigAbBeforeGueltigBis(
+                                                zuordnung.gueltigAb,
+                                                zuordnung.gueltigBis,
+                                                'Datum ab muss <= Datum bis.'
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:122]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 7]
```diff
@@ -61,12 +61,13 @@
                     <v-text-field
                         ref="address"
                         v-model="zuordnung.funktionsadresse"
-                        label="Funktionsadresse"
+                        label="EMail"
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:64]: ✅

File: mobidam-sst-management-backend/src/main/java/de/muenchen/mobidam/domain/Zuordnung.java [hunk 8]
```diff
@@ -75,9 +75,6 @@ public class Zuordnung extends BaseEntity {
     @FutureOrPresent
     private LocalDate gueltigAb;
 
-    @Column(nullable = false)
-    @NotNull
-    @FutureOrPresent
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-backend/src/main/java/de/muenchen/mobidam/domain/Zuordnung.java:80]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 9]
```diff
@@ -196,9 +208,36 @@ const form = ref<VForm>();
...
-    if (valid) {
-        emit("zuordnung-saved", zuordnung.value);
-        closeDialog();
+    {
+        if (valid) {
+            if (
+                isGueltigAbBeforeGueltigBis(
+                    zuordnung.value.gueltigAb,
+                    zuordnung.value.gueltigBis,
+                    "Datum ab muss <= Datum bis."
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:217]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 10]
```diff
@@ -196,9 +208,36 @@ const form = ref<VForm>();
...
+                closeDialog();
+            }
+        }
+    }
+}
+
+function isGueltigAbBeforeGueltigBis(
+    fromDate: string,
+    toDate: string | number | Date | undefined,
+    message: string
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:230]: I tried that before without success. Now it works. Apparently I did something wrong at the time. ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 11]
```diff
@@ -196,9 +208,36 @@ const form = ref<VForm>();
 async function saveTask() {
     // eslint-disable-next-line no-unsafe-optional-chaining
     const valid = (await form.value?.validate())?.valid;
-    if (valid) {
-        emit("zuordnung-saved", zuordnung.value);
-        closeDialog();
+    {
+        if (valid) {
+            if (
+                isGueltigAbBeforeGueltigBis(
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:214]: ✅ There was a reason why I added it. After removing the additional check and testing everything worked as planned. The extended check is therefore superfluous and has been removed again.

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 12]
```diff
@@ -196,9 +208,36 @@ const form = ref<VForm>();
 async function saveTask() {
     // eslint-disable-next-line no-unsafe-optional-chaining
     const valid = (await form.value?.validate())?.valid;
-    if (valid) {
-        emit("zuordnung-saved", zuordnung.value);
-        closeDialog();
+    {
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:211]: ✅ The additional check is removed. No more unnecessary brackets.

File: mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue [hunk 13]
```diff
@@ -224,7 +234,13 @@ function updateSchnittstelle() {
         for (const zuordnung of mutableZuordnungen.value) {
             if (!dialogProps.zuordnungen.includes(zuordnung)) {
                 zuordnung.schnittstelle = dialogProps.schnittstelle.id;
-                await ZuordnungService.create(zuordnung);
+                try {
+                    await ZuordnungService.create(zuordnung);
+                } catch (error) {
+                    HealthService.checkHealth().catch((error) => {
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue:240]: Without it, there is no indication of an error while saving.

File: mobidam-sst-management-frontend/frontend/src/composables/rules.ts [hunk 14]
```diff
@@ -37,9 +37,26 @@ export function useRules() {
...
+            return false;
+        }
+        const today = new Date();
+        const to = new Date(toDate);
+        return to < today;
+    }
+
     return {
         maxLengthRule,
         notEmptyDateRule,
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/composables/rules.ts:57]: ✅ Removed.

File: mobidam-sst-management-frontend/frontend/src/views/MainView.vue (hunk 3 above)
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/views/MainView.vue:68]: Maybe i misunderstand this, but there is another colum.
image

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue [hunk 15]
```diff
@@ -61,12 +61,13 @@
                     <v-text-field
                         ref="address"
                         v-model="zuordnung.funktionsadresse"
-                        label="Funktionsadresse"
+                        label="Funktionspostfach"
                         hint="Welchem Gruppenpostfach gehört diese Person an?"
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:65]: Wie lautet das Funktionspostfach?

File: mobidam-sst-management-frontend/frontend/src/composables/rules.ts [hunk 16]
```diff
@@ -27,19 +27,48 @@ export function useRules() {
...
 
-    function notEmptyRule(message = "error") {
+    function isValidEmail(message = "error") {
+        const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
         return (value: string | null | undefined) =>
-            (value && value.trim() != "") || message;
+            (value && emailPattern.test(value.trim())) || message;
+    }
+
+    function isExpiredGueltigBis(toDate: string | undefined) {
```
Comment (martind260) [inline mobidam-sst-management-frontend/frontend/src/composables/rules.ts:41]: No need to use specifier, just name it: function isExpired(date: string | undefined) {

File: mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue (hunk 13 above)
[the remark this answers is not in the document]
Comment (martind260) [reply mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue:240]: You already have caught the error when you call ZuordnungService.create(). You want to show exactly this error in the snackbar. You dont want to show the error that comes from HealthService, right?

File: mobidam-sst-management-frontend/frontend/src/composables/rules.ts (hunk 16 above)
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/composables/rules.ts:41]: ✅

File: mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue (hunk 15 above)
Comment (sfi2022) [reply mobidam-sst-management-frontend/frontend/src/components/AddPersonDialog.vue:65]: This means _hint="Wie lautet das Funktionspostfach?"_   ?

File: mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue (hunk 13 above)
[the remark this answers is not in the document]
Comment (martind260) [reply mobidam-sst-management-frontend/frontend/src/components/ManageSchnittstelleDialog.vue]: Seems to be a bigger problem in error handling. Will be treated outside of this issue.

**Review by martind260 — APPROVED** [review]

**Pull request merged**